api/server/server.go

```diff
@@ -374,42 +374,31 @@ func getContainersTop(eng *engine.Engine, version version.Version, w http.Respon
 }
 
 func getContainersJSON(eng *engine.Engine, version version.Version, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	if err := parseForm(r); err != nil {
+	var err error
+	if err = parseForm(r); err != nil {
 		return err
 	}
-	var (
-		err  error
-		outs *engine.Table
-		job  = eng.Job("containers")
-	)
-
-	job.Setenv("all", r.Form.Get("all"))
-	job.Setenv("size", r.Form.Get("size"))
-	job.Setenv("since", r.Form.Get("since"))
-	job.Setenv("before", r.Form.Get("before"))
-	job.Setenv("limit", r.Form.Get("limit"))
-	job.Setenv("filters", r.Form.Get("filters"))
 
-	if version.GreaterThanOrEqualTo("1.5") {
-		streamJSON(job, w, false)
-	} else if outs, err = job.Stdout.AddTable(); err != nil {
-		return err
-	}
-	if err = job.Run(); err != nil {
-		return err
+	config := daemon.ContainersConfig{
+		All:     r.Form.Get("all") == "1",
+		Size:    r.Form.Get("size") == "1",
+		Since:   r.Form.Get("since"),
+		Before:  r.Form.Get("before"),
+		Filters: r.Form.Get("filters"),
 	}
-	if version.LessThan("1.5") { // Convert to legacy format
-		for _, out := range outs.Data {
-			ports := engine.NewTable("", 0)
-			ports.ReadListFrom([]byte(out.Get("Ports")))
-			out.Set("Ports", api.DisplayablePorts(ports))
-		}
-		w.Header().Set("Content-Type", "application/json")
-		if _, err = outs.WriteListTo(w); err != nil {
+
+	if tmp := r.Form.Get("limit"); tmp != "" {
+		if config.Limit, err = strconv.Atoi(tmp); err != nil {
```

Comment:
Sorry I just don’t see the point in a new variable when I think the code is clear as it is.

> On Apr 8, 2015, at 1:45 PM, Alexander Morozov <notifications@github.com> wrote:
> 
> In api/server/server.go <https://github.com/docker/docker/pull/12163#discussion_r27993999>:
> 
> >  	}
> > -	if version.LessThan("1.5") { // Convert to legacy format
> > -		for _, out := range outs.Data {
> > -			ports := engine.NewTable("", 0)
> > -			ports.ReadListFrom([]byte(out.Get("Ports")))
> > -			out.Set("Ports", api.DisplayablePorts(ports))
> > -		}
> > -		w.Header().Set("Content-Type", "application/json")
> > -		if _, err = outs.WriteListTo(w); err != nil {
> > +
> > +	if tmp := r.Form.Get("limit"); tmp != "" {
> > +		if config.Limit, err = strconv.Atoi(tmp); err != nil {
> Point is not to use outer scope err. Local scope always better reads, and var err error is always scary, because you need to look through whole code to see for what it needed - sometimes it is necessary, but not here.
> Also all over the code used val, err := func() pattern, and I prefer to use less globalVar, globalErr = func() pattern.
> Another style concern that config.Limit is global and err local by logic and mixing them in same assignment sorta tricky.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/12163/files#r27993999>.
>